zerver/webhooks/patreon/view.py

```diff
@@ -0,0 +1,205 @@
+        f"members:delete, where last_charge_status={last_charge_status}"
+    )
+
+
+def get_members_pledge_create_body(payload: WildValue) -> str:
+    last_charge_status = get_last_charge_status(payload)
+    pledge_amount = get_pledge_amount(payload)
+    # The only successful charge status is "Paid". null if not yet charged.
+    if last_charge_status in ("Paid", None) and pledge_amount > 0:
+        template = "{user_name} has pledged ${pledge_amount:.2f} per {pay_per_name}. :tada:\nTotal number of patrons: {patron_count}"
+        return template.format(
+            user_name=get_user_name(payload),
+            pledge_amount=pledge_amount,
+            pay_per_name=get_pay_per_name(payload),
+            patron_count=get_patron_count(payload),
+        ).rstrip()
+    raise UnsupportedWebhookEventTypeError(
+        f"members:pledge:create, where last_charge_status={last_charge_status}"
+    )
+
+
+def get_members_pledge_update_body(payload: WildValue) -> str:
+    last_charge_status = get_last_charge_status(payload)
+    pledge_amount = get_pledge_amount(payload)
+    # The only successful charge status is "Paid". null if not yet charged.
+    if last_charge_status in ("Paid", None) and pledge_amount > 0:
+        template = "{user_name} has updated their pledge to ${pledge_amount:.2f} per {pay_per_name}. :gear:"
+        return template.format(
+            user_name=get_user_name(payload),
+            pledge_amount=pledge_amount,
+            pay_per_name=get_pay_per_name(payload),
+        ).rstrip()
+    raise UnsupportedWebhookEventTypeError(
+        f"members:pledge:update, where last_charge_status={last_charge_status}"
+    )
+
+
+def get_members_pledge_delete_body(payload: WildValue) -> str:
+    last_charge_status = get_last_charge_status(payload)
+    if last_charge_status in ("Paid", "Deleted", None):
+        template = "{user_name}'s pledge has been cancelled. :cross_mark:\nTotal number of patrons: {patron_count}"
+        return template.format(
+            user_name=get_user_name(payload),
+            patron_count=get_patron_count(payload),
+        ).rstrip()
+    raise UnsupportedWebhookEventTypeError(
+        f"members:pledge:delete, where last_charge_status={last_charge_status}"
+    )
+
+
+def get_user_name(payload: WildValue) -> str:
+    return payload["data"]["attributes"]["full_name"].tame(check_string)
+
+
+def get_pledge_amount(payload: WildValue) -> float:
+    return payload["data"]["attributes"]["currently_entitled_amount_cents"].tame(check_int) / 100
+
+
+def get_patron_count(payload: WildValue) -> int:
+    return payload["included"][0]["attributes"]["patron_count"].tame(check_int)
+
+
+def get_pay_per_name(payload: WildValue) -> str:
+    return payload["included"][0]["attributes"]["pay_per_name"].tame(check_string)
+
+
+def get_last_charge_status(payload: WildValue) -> object:
+    return payload["data"]["attributes"]["last_charge_status"].tame(check_anything)
```

Comment:
Should this be `.tame(check_none_or(check_string))` instead of checking for anything? Same not for `get_patron_status`.

Reply:
Yes, you are right! I just didn't realize there was a `check_none_or` function I could use. Thanks!